### PR TITLE
Add implementation for Precision/Recall, is_multilabel=True, average=True

### PR DESCRIPTION
**Pull request by anmolsjoshi**
Related to #417 

Description:
Added implementation that does not keep entire training history when average=True for multilabel cases. 

Added tests to check whether Precision(is_multilabel=True, average=True) equals Precision(is_multilabel=True, average=False).mean()

Check list:
* [x] New tests are added (if a new feature is modified)
* [ ] New doc strings: text and/or example code are in RST format
* [ ] Documentation is updated (if required)

### CODE REVIEW

**Comment by anmolsjoshi** [comment]
@vfdev-5 I have fixed the flake8 errors in the two latest commits. Are we good to merge?

**Comment by vfdev-5** [comment]
@anmolsjoshi please merge the master into the branch

**Comment by anmolsjoshi** [comment]
@vfdev-5 done! please review and merge.

File: ignite/metrics/precision.py [hunk 1]
```diff
@@ -15,11 +15,11 @@ def __init__(self, output_transform=lambda x: x, average=False, is_multilabel=Fa
...
 
     def reset(self):
-        self._true_positives = torch.DoubleTensor(0) if self._is_multilabel else 0
-        self._positives = torch.DoubleTensor(0) if self._is_multilabel else 0
+        self._true_positives = torch.DoubleTensor(0) if (self._is_multilabel and not self._average) else 0
+        self._positives = torch.DoubleTensor(0) if (self._is_multilabel and not self._average) else 0
 
     def compute(self):
-        if not isinstance(self._positives, torch.Tensor):
+        if not (isinstance(self._positives, torch.Tensor) or self._positives > 0):
```
Comment (vfdev-5) [inline ignite/metrics/precision.py]: @anmolsjoshi could you please add a test to raise `NotComputableError` in two cases of multilabel. I think we missed `or self._positives > 0` in the current implementation

**Comment by vfdev-5** [comment]
Looks good, just a small test is needed.

File: ignite/metrics/precision.py (hunk 1 above)
Comment (anmolsjoshi) [reply ignite/metrics/precision.py]: Done!

**Comment by vfdev-5** [comment]
@anmolsjoshi thanks !

**Pull request merged**